File: JavaScript/1.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Digite o comprimento do retângulo ou lado do quadrado: ', (comprimento) => {
    rl.question('Digite a largura do retângulo (ou pressione Enter para usar o mesmo valor do comprimento): ', (largura) => {
        comprimento = parseFloat(comprimento);
        largura = largura.trim() === '' ? NaN : parseFloat(largura);

        let perimetro;
        if (isNaN(largura)) {
            perimetro = comprimento * 4;
            console.log("O perímetro do quadrado é: ", perimetro);
        } else {
            perimetro = 2 * (comprimento + largura);
            console.log("O perímetro do retângulo é: ", perimetro);
        }

        rl.close();
    });
});

File: Python/1.py
comprimento = float(input("Digite o comprimento do retângulo ou lado do quadrado: "))
largura_input = input("Digite a largura do retângulo (ou pressione Enter para usar o mesmo valor do comprimento): ")

if largura_input == "":
    largura = comprimento
else:
    largura = float(largura_input)

if largura == comprimento:
    perimetro = comprimento * 4
    print("O perímetro do quadrado é:", perimetro)
else:
    perimetro = 2 * (comprimento + largura)
    print("O perímetro do retângulo é:", perimetro)
